app/Services/AdminService.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace App\Services;
+
+use App\Mail\AdminPasswordMail;
+use App\Models\Admin;
+use App\Repositories\AdminRepository;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Mail;
+use Illuminate\Support\Str;
+use Spatie\Permission\Models\Permission;
+
+class AdminService extends BaseService
+{
+    public const is_super = 0;
+
+    public function __construct(AdminRepository $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    public function handleData($data)
+    {
+        $password = Str::random(10);
```

Comment:
Creating a password of admin could be put as a mutator in the model
https://laravel.com/docs/8.x/eloquent-mutators#defining-a-mutator